x-pack/legacy/plugins/ml/public/application/services/ml_api_service/data_frame_analytics.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { http } from '../http_service';
+
+import { basePath } from './index';
+import { DataFrameAnalyticsStats } from '../../data_frame_analytics/pages/analytics_management/components/analytics_list/common';
+import { DataFrameAnalyticsConfig } from '../../data_frame_analytics/common';
+
+export interface GetDataFrameAnalyticsStatsResponseOk {
+  node_failures?: object;
+  count: number;
+  data_frame_analytics: DataFrameAnalyticsStats[];
+}
+
+export interface GetDataFrameAnalyticsStatsResponseError {
+  statusCode: number;
+  error: string;
+  message: string;
+}
+
+export type GetDataFrameAnalyticsStatsResponse =
+  | GetDataFrameAnalyticsStatsResponseOk
+  | GetDataFrameAnalyticsStatsResponseError;
+
+interface GetDataFrameAnalyticsResponse {
+  count: number;
+  data_frame_analytics: DataFrameAnalyticsConfig[];
+}
+
+export const dataFrameAnalytics = {
+  getDataFrameAnalytics(analyticsId?: string) {
+    const analyticsIdString = analyticsId !== undefined ? `/${analyticsId}` : '';
+    return http<GetDataFrameAnalyticsResponse>({
+      path: `${basePath()}/data_frame/analytics${analyticsIdString}`,
+      method: 'GET',
+    });
+  },
+  getDataFrameAnalyticsStats(analyticsId?: string) {
+    if (analyticsId !== undefined) {
+      return http<GetDataFrameAnalyticsStatsResponse>({
+        path: `${basePath()}/data_frame/analytics/${analyticsId}/_stats`,
+        method: 'GET',
+      });
+    }
+
+    return http<GetDataFrameAnalyticsStatsResponse>({
+      path: `${basePath()}/data_frame/analytics/_stats`,
+      method: 'GET',
+    });
+  },
+  createDataFrameAnalytics(analyticsId: string, analyticsConfig: any) {
```

Comment:
You could use `DataFrameAnalyticsConfig` from `data_frame_analytics/common/analytics.ts` maybe?

Reply:
the places where `createDataFrameAnalytics` is called aren't supplying a `DataFrameAnalyticsConfig` object. 
It looks like a `DeepPartial<DataFrameAnalyticsConfig>`.
I've updated the expected type to be  `DeepPartial<DataFrameAnalyticsConfig>`.